File: src/main/java/net/softbell/bsh/dto/view/InterlockTokenCardDto.java
package net.softbell.bsh.dto.view;

import java.util.Date;

import lombok.Getter;
import lombok.Setter;
import net.softbell.bsh.domain.EnableStatusRule;
import net.softbell.bsh.domain.entity.MemberInterlockToken;

/**
 * @Author : Bell(bell@softbell.net)
 * @Description : 회원 연동 토큰뷰 카드정보 DTO
 */
@Getter
@Setter
public class InterlockTokenCardDto
{
	private Long tokenId;
	private String name;
	private String token;
	private EnableStatusRule enableStatus;
	private Date registerDate;
	
	public InterlockTokenCardDto(MemberInterlockToken entity)
	{
		// Exception
		if (entity == null)
			return;
		
		// Convert
		this.tokenId = entity.getMemberInterlockId();
		this.name = entity.getName();
		this.token = entity.getToken();
		this.enableStatus = entity.getEnableStatus();
		this.registerDate = entity.getRegisterDate();
	}
}

File: src/main/kotlin/net/softbell/bsh/dto/view/InterlockTokenCardDto.kt
package net.softbell.bsh.dto.view

import net.softbell.bsh.domain.EnableStatusRule
import net.softbell.bsh.domain.entity.MemberInterlockToken
import java.util.*

/**
 * @author : Bell(bell@softbell.net)
 * @description : 회원 연동 토큰뷰 카드정보 DTO
 */
class InterlockTokenCardDto(entity: MemberInterlockToken) {
    val tokenId: Long = entity.memberInterlockId
    val name: String = entity.name
    val token: String = entity.token
    val enableStatus: EnableStatusRule = entity.enableStatus
    val registerDate: Date = entity.registerDate
}